CODINGSTYLE.md

```diff
@@ -14,13 +14,57 @@ In these cases, it can be overridden by surrounding it with `clang-format-off` a
 
 ### Naming convention
 
-* Almost everything uses `lower_snake_case`.
+* Local variables uses `lower_snake_case`.
 * `#define` uses `UPPER_SNAKE_CASE`.
 * Everything is in a namespace.
 * Therefore, there are no prefixes.
-* Private data members have an "m" prefix: `m_data`.
-* Getter functions use the name without "m": `data()`.
-* Setter functions use the "set" prefix: `set_data(value)`.
+
+### Objects types
+
+#### Structs
+
+Structs shall only be used for "plain-old-data" POD type with no logic or any functions except a data member initialization function (constructor).
+
+* Struct definition shall use camel case with `s` prefix: `sSomeStruct`.
```

Comment:
The main reason I did it now is since you gave a comment about it on another PR of mine. 😄 
But besides that, I think that is missing and we need a convention for that.
Using the same convention for type and variables name is annoying IMO since usually we would want to give the same name of the type to the variable itself.

Bad IMO:
```cpp
class some_class {
    // code
};

// This is annoying since it forces me to be creative and give the variable a different name from the type
some_class some_class_obj;
```

Good IMO
```cpp
class SomeClass {
    // code
};

// Perfect IMO :)
SomeClass some_class;
```

Reply:
I tend to agree with @morantr on this one now, at least new classes should use camle case for the class definition.